### DISCUSSION TITLE
2.0版本只支持react 18以上了吗

### DISCUSSION THREAD

**Discussion by JackLLLLL** [antvis/X6#2939] [Q&A] [answered]
最近看到有新版本, 尝试升级, 发现原来的react 17已经不行了会报错, react 18以上版本才可以运行 

如果是这样的话是不是最好官网在”升到2.x“文档中写一下~

**Comment by NewByVector** [comment]
我们的疏忽，现在 x6-react-shape 2.x 包只支持 react18 了。

**Comment by NewByVector** [comment, marked as answer]
号外号外，在 @antv/x6-react-shape 2.0.5 版本中同时支持了 react17 和 react18

**Reply by JackLLLLL** [reply]
请问需要额外配置什么东西吗？把react-shape的版本升上去以后还是会提示 Can't resolve 'react-dom/client'
![image](https://user-images.githubusercontent.com/37646584/206963472-ae286b65-f931-416a-aafa-5deef43ba928.png)

**Reply by NewByVector** [reply]
确实，在一些框架下对 require 内容是需要编译检测的，我再思考下有没有好的方法来兼容，如果是在不行，我们会在 2.0.x 支持 react17, 在 2.1.x 中不再支持 react17。

**Reply by NewByVector** [reply]
x6-react-shape@2.0.8 已经支持 react17 了。

**Comment by JackLLLLL** [comment]
awesome !
